### ISSUE TITLE
Add new algorithm: SHA512 for FileZillaServer

### ISSUE THREAD

**Issue by NeoTheCapt** [hashcat/hashcat#943]
example hash:
hash:
8109F61B7CED2604BA7ACA37F96B276C910E5F8BD711FB9629628C8AF32D6B9D9071FF5A4A28650A7B7703EFB0ACB52372BAF696626E106495F8D328602E4716
salt:
o}9Dh`/XH?b!$f:Dtv$Z=VFlzS+O@it1GcH/nt@S4378z-xhltD94<c:wawEE~FV

FileZillaServer comes with SHA512 hash,but the salt length is fixed in 64.

**Comment by kholia** [comment]
The algorithm is simply sha512($p.$s). The password for this particular hash is "wyywyy".

https://github.com/magnumripper/JohnTheRipper/pull/2374 is relevant to this topic.

**Comment by NeoTheCapt** [comment]
This is Hashcat dude,not JohnTheRipper.
Hashcat wont give a fuk on salt like 64 bytes length.

**Comment by jsteube** [comment]
@usualwyy I think what @kholia wanted to say is that with JtR you can already crack this hash because it supports 64 bytes length (and more) salts.

For hashcat we need a special kernel to do that, therefore it's correct to open up this issue,

**Comment by NeoTheCapt** [comment]
@jsteube Yes,we got solutions on cracking the long length salt,like EGB,JTR bla bla bla.They can do the job,I knew that.
But none of them as good as hashcat,that's why Im here asking for a new algorithm.

**Comment by magnumripper** [comment]
> that's why Im here asking for a new algorithm.

And that's why @kholia pointed to some additional samples and example parsing code just in case it would spare Atom some work or research. You sound like a dumb kid.

**Comment by kost123** [comment]
+1 for this, longer salt length strikes again :D

**Comment by philsmd** [comment]
Some additional (nice-to-know) notes about the FileZilla Server algorithm (for devs):
the FileZilla Server source code is available here:
` svn co https://svn.filezilla-project.org/svn/FileZilla%20Server/trunk filezilla_server`

It seems that FileZilla Server used to MD5 ($pass) hash the user credentials in versions < v0.9.55, see 
https://svn.filezilla-project.org/filezilla/FileZilla%20Server/trunk/readme.htm?revision=7789&content-type=text%2Fplain : 

> Newly set account passwords are now stored in form of salted SHA512 hashes

Furthermore the SaveUser () function seems to be interesting:
https://svn.filezilla-project.org/filezilla/FileZilla%20Server/trunk/source/Interface/UsersDlgGeneral.cpp?view=markup#l265
and of course the generateSalt () function:
https://svn.filezilla-project.org/filezilla/FileZilla%20Server/trunk/source/Accounts.cpp?view=markup#l471

To sum it up, the salt length seems to be hard-coded (to 64 bytes exactly) and the first version that supports this is v0.9.55, so we might add a format like -m 1712 = FileZilla Server >= v0.9.55

**Comment by kost123** [comment]
Don't suppose this is easily upgradable to 128 salt length? not related to filezilla, just for an obscure software, but if its easy enough I'd just do it for myself.
